compiler_opt/tools/generate_default_trace.py

```diff
@@ -97,35 +99,41 @@ def worker(policy_path: str, work_queue: 'queue.Queue[corpus.ModuleSpec]',
     results_queue: the queue where results are deposited.
     key_filter: regex filter for key names to include, or None to include all.
   """
-  runner = get_runner()
-  m = re.compile(key_filter) if key_filter else None
-
-  while True:
-    try:
-      module_spec = work_queue.get_nowait()
-    except queue.Empty:
-      return
-    try:
-      data = runner.collect_data(
-          module_spec=module_spec, tf_policy_path=policy_path, reward_stat=None)
-      if not m:
-        results_queue.put((module_spec.name, data.serialized_sequence_examples,
-                           data.reward_stats))
-        continue
-      new_reward_stats = {}
-      new_sequence_examples = []
-      for k, sequence_example in zip(data.keys,
-                                     data.serialized_sequence_examples):
-        if not m.match(k):
+  try:
+    runner = get_runner()
+    m = re.compile(key_filter) if key_filter else None
+
+    while True:
+      try:
+        module_spec = work_queue.get_nowait()
+      except queue.Empty:
+        return
+      try:
+        data = runner.collect_data(
+            module_spec=module_spec,
+            tf_policy_path=policy_path,
+            reward_stat=None)
+        if not m:
+          results_queue.put(
+              (module_spec.name, data.serialized_sequence_examples,
+               data.reward_stats))
           continue
-        new_reward_stats[k] = data.reward_stats[k]
-        new_sequence_examples.append(sequence_example)
-      results_queue.put(
-          (module_spec.name, new_sequence_examples, new_reward_stats))
-    except (subprocess.CalledProcessError, subprocess.TimeoutExpired,
-            RuntimeError):
-      logging.error('Failed to compile %s.', module_spec.name)
-      results_queue.put(None)
+        new_reward_stats = {}
+        new_sequence_examples = []
+        for k, sequence_example in zip(data.keys,
+                                      data.serialized_sequence_examples):
+          if not m.match(k):
+            continue
+          new_reward_stats[k] = data.reward_stats[k]
+          new_sequence_examples.append(sequence_example)
+        results_queue.put(
+            (module_spec.name, new_sequence_examples, new_reward_stats))
+      except (subprocess.CalledProcessError, subprocess.TimeoutExpired,
+              RuntimeError):
+        logging.error('Failed to compile %s.', module_spec.name)
+        results_queue.put(None)
```

Comment:
why not have the except her?

Reply:
i mean do not really need another try except outside, just have try except except here to queue the error to result queue

Reply:
hmm...actually it is a bit tricky. so if the compilation fails due to some flag issue (the case you'd want to capture and kill early), it is going to be one of the subprocess.CalledProcessError, subprocess.TimeoutExpired,RuntimeError so it won't raise the BaseException.

one alternative just to have the receiver check on how many None it receives. for whatever reason, if it receives "too much" Nones, e.g., 50%, it kills the code

Reply:
This is not about that. The compilation error cases are treated together and a report is generated. This is about a bug like we change parameter names in compilation_runner.

basically, compilation failures are expected - timeouts, one-off issue in the corpus, whatever. If the user sees what is unacceptable to them as a number of failed compilation tasks, they can investigate.

bugs like I mentioned are not expected. There's no `analog` thing there (i.e. degrees of allowance), it's straight-out failure.

Reply:
ic, so essentially you are only trying to catch the error in 'runner = get_runner()', it makes sense then.

i'm wondering why the get_runner() is moved to inside the multiprocessing?